weatherman.py

```diff
@@ -0,0 +1,177 @@
+        dayscount += 1
+        line_count += 1
+  return dayslist
+
+def finding_values1(list):
+  maxtemp = -9999999
+  mintemp = 9999999
+  maxmeanhumidity = -9999999
+  daycount = 0
+  for day in list:
+    if(day.maxtemp != ""):
+      if(int(day.maxtemp) > maxtemp):
+        maxtemp = int(day.maxtemp)
+        maxtempday = daycount
+    if(day.mintemp != ""):
+      if(int(day.mintemp) < mintemp):
+        mintemp = int(day.mintemp)
+        mintempday = daycount
+    if(day.meanhumidity != ""):
+      if(int(day.meanhumidity) > maxmeanhumidity):
+        maxmeanhumidity = int(day.meanhumidity)
+        maxhumidday = daycount
+    daycount += 1
+  result = [list[maxtempday], list[mintempday], list[maxhumidday]]
+  return result
+
+def finding_values2(list):
+  avgmaxtemp = 0
+  avgmintemp = 0
+  avgmaxmeanhumidity = 0
+  daycount = 0
+  for day in list:
+    if(day.maxtemp != ""):
+      avgmaxtemp += int(day.maxtemp)
+    if(day.mintemp != ""):
+      avgmintemp += int(day.mintemp)
+    if(day.meanhumidity != ""):
+      avgmaxmeanhumidity += int(day.meanhumidity)
+    daycount += 1
+  avgmaxtemp /= daycount
+  avgmintemp /= daycount
+  avgmaxmeanhumidity /= daycount
+  result = [[avgmaxtemp], [avgmintemp], [avgmaxmeanhumidity]]
+  return result
+
+
+def display1(result):
+  print("The max temprature is:", result[0].maxtemp, "on", result[0].date)
+  print("The min temprature is:", result[1].mintemp, "on", result[1].date)
+  print("The most humid day is:", result[2].meanhumidity, "on", result[2].date)
+
+def display2(result):
+  print("The avg max temprature is:", result[0])
+  print("The avg min temprature is:", result[1])
+  print("The avg humidity is:", result[2])  
+
+def display3(result):
+  daycount = 1
+  for day in result:
+    print(daycount, ":")
+    count1 = 0
+    count2 = 0
+    if(day.maxtemp != ""):
+      count1 = int(day.maxtemp)
+    if(day.mintemp != ""):
+      count2 = int(day.mintemp)
+    if(count1 != 0):
+      for i in range(count1):
+        print("+", end = " ")
+      print(count1, "C")
+    if(count2 != 0):
+      for i in range(count2):
+        print("+", end = " ")
+      print(count2, "C\n\n")
+    daycount += 1
+    
+
+def parse(days_list):
+  parser = argparse.ArgumentParser()
+  parser.add_argument("path", type = str, help = "path to weather files")
+  parser.add_argument("-e", "--year", type = int, help = "Insert year to see its weather extremes", nargs = "+")
+  parser.add_argument("-a", "--date1", type = str, help = "Insert year with its month to see its weather extremes", nargs = "+")
```

Comment:
instead of type str/int, use custom function that should convert input into datetime object and raise error if input is not valid